addons/stock/models/stock_move.py

```diff
@@ -510,6 +510,17 @@ def write(self, vals):
             move_to_recompute_state._recompute_state()
         if receipt_moves_to_reassign:
             receipt_moves_to_reassign._action_assign()
+        # when assign picking for inter-warehouse transfers, set the warehouses as partners on pickings
```

Comment:
note : if you set the partner_id in the route you need this code.

note : you should juste loop on each picking (dans not on each move).